### DISCUSSION TITLE
Optimizing Prefetch in Next.js after URL Manipulation via CDN

### DISCUSSION THREAD

**Discussion by tavareshenrique** [vercel/next.js#62327] [Help]
### Summary

I have a route in my application named `slug`.

This route is dynamic, so I can have:

https://my-site.com/slug/test-1
https://my-site.com/slug/test-2
https://my-site.com/slug/new-article

However, within a `CDN` that I use, I make the necessary adjustments so that this `/slug/` route does not appear in the `URL`, thus becoming:

https://my-site.com/test-1
https://my-site.com/test-2
https://my-site.com/new-article

Within each of these pages, there are links and navigation buttons to other pages of the application.

With that, I want to use the prefetch feature of `Next/Link` to make the loading of these pages faster, aiming to improve the INP.

However, due to this change I make via CDN, NextJS cannot prefetch the pages.

Assuming I am on the route:

https://my-site.com/test-1

Inside it, I have a link that leads to https://my-site.com/test-2

If I use prefetch on this link, it's as if nothing happens.

Is there any way to fix this issue?

### Additional information

Just to clarify, unfortunately, these route adjustments via CDN are necessary. I cannot forgo this at the moment, as there are thousands of other routes in the application.

**Comment by gustavostz** [comment]
Since your CDN is changing URLs, Next.js can't prefetch pages automatically. You can manually prefetch pages using Next.js' router. For example, when a component mounts or when a user hovers over a link, you can prefetch the page like this:

```
import { useEffect } from 'react';
import { useRouter } from 'next/router';

const MyComponent = () => {
  const router = useRouter();

  useEffect(() => {
    // Prefetch the page
    router.prefetch('/slug/test-2');
  }, []);

  return (
    <div>
      <a href="/test-2" onClick={(e) => {
        e.preventDefault();
        router.push('/slug/test-2');
      }}>
        Go to Test 2
      </a>
    </div>
  );
};

export default MyComponent;
```

In this code, we're manually prefetching the page /slug/test-2 when the component mounts. When the link is clicked, we use router.push to navigate to the page, ensuring that the prefetched page is used. This way, you can bypass the issue caused by the CDN's URL changes.

**Reply by tavareshenrique** [reply]
Thank you so much, @gustavostz.

I will test it and bring feedback later.
